vm/actor/src/builtin/miner/types.rs

```diff
@@ -9,105 +9,179 @@ use clock::ChainEpoch;
 }
+
 #[derive(Serialize_tuple, Deserialize_tuple)]
 pub struct ChangePeerIDParams {
     #[serde(with = "serde_bytes")]
     pub new_id: Vec<u8>,
 }
+
 #[derive(Serialize_tuple, Deserialize_tuple)]
 pub struct ChangeMultiaddrsParams {
     #[serde(with = "serde_bytes")]
     pub new_multi_addrs: Vec<u8>,
 }
+
 #[derive(Serialize_tuple, Deserialize_tuple)]
 pub struct ConfirmSectorProofsParams {
     pub sectors: Vec<SectorNumber>,
 }
+
+#[derive(Serialize_tuple, Deserialize_tuple)]
+pub struct PoStPartition {
+    /// Partitions are numbered per-deadline, from zero.
+    pub index: u64,
+    /// Sectors skipped while proving that weren't already declared faulty.
+    pub skipped: BitField,
+}
+
 /// Information submitted by a miner to provide a Window PoSt.
 #[derive(Serialize_tuple, Deserialize_tuple)]
 pub struct SubmitWindowedPoStParams {
     /// The deadline index which the submission targets.
     pub deadline: u64,
-    /// The partition indices being proven.
-    /// Partitions are counted across all deadlines, such that all partition indices in the second deadline are greater
-    /// than the partition numbers in the first deadlines.
-    pub partitions: Vec<u64>,
+    /// The partitions being proven.
+    pub partitions: Vec<PoStPartition>,
     /// Array of proofs, one per distinct registered proof type present in the sectors being proven.
     /// In the usual case of a single proof type, this array will always have a single element (independent of number of partitions).
     pub proofs: Vec<PoStProof>,
-    /// Sectors skipped while proving that weren't already declared faulty
-    pub skipped: BitField,
+    /// The epoch at which these proofs is being committed to a particular chain.
+    pub chain_commit_epoch: ChainEpoch,
+    /// The ticket randomness on the chain at the `chain_commit_epoch` on the chain this post is committed to.
+    pub chain_commit_rand: Randomness,
 }
+
 #[derive(Serialize_tuple, Deserialize_tuple)]
 pub struct ProveCommitSectorParams {
     pub sector_number: SectorNumber,
     #[serde(with = "serde_bytes")]
     pub proof: Vec<u8>,
 }
+
 #[derive(Serialize_tuple, Deserialize_tuple)]
 pub struct CheckSectorProvenParams {
     pub sector_number: SectorNumber,
 }
+
 #[derive(Serialize_tuple, Deserialize_tuple)]
 pub struct ExtendSectorExpirationParams {
-    pub sector_number: SectorNumber,
+    pub extensions: Vec<ExpirationExtension>,
+}
+
+#[derive(Serialize_tuple, Deserialize_tuple)]
+pub struct ExpirationExtension {
+    pub deadline: u64,
+    pub partition: u64,
+    pub sectors: BitField,
     pub new_expiration: ChainEpoch,
 }
+
 #[derive(Serialize_tuple, Deserialize_tuple)]
 pub struct TerminateSectorsParams {
+    pub terminations: Vec<TerminationDeclaration>,
+}
+
+#[derive(Serialize_tuple, Deserialize_tuple)]
+pub struct TerminationDeclaration {
+    pub deadline: u64,
+    pub partition: u64,
     pub sectors: BitField,
 }
+
+pub struct TerminateSectorsReturn {
```

Comment:
This type should be Serialize/deserialize. I noticed you are not using the return for the `terminate_sectors` function. I will comment on that when I get to that file, but mentioning now

Reply:
Ahh yep, got it